### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by TheGeorge**
Added some documentation and added codecov.io (for trying out)

### CODE REVIEW

**Comment by TheGeorge** [comment]
Travis is not running on the pull request. I guess that this needs to be enabled after moving the repository.

**Comment by kostis** [comment]
Travis has been enabled on the new repo, but the build is failing miserably.
More concerning is the fact that the failure does not seem to be related to documentation changes but to `pip` which I do not really see what it has to do with building documentation.

File: .travis.yml [hunk 1]
```diff
@@ -15,3 +15,8 @@ otp_release:
 cache:
    directories:
    - .plt
+
+install:
+    pip install codecov
```
Comment (fenollp) [inline .travis.yml]: How about `bash <(curl -s https://codecov.io/bash)` instead?

**Comment by TheGeorge** [comment]
@fenollp I was considering that as well. The codecov example for Erlang (
https://github.com/codecov/example-erlang) from their documentation states
to do it like in this pr.

Maybe it's best to divide the pr into the documentation update and make a
separate one for getting codecov to work.

**Comment by kostis** [comment]
> Maybe it's best to divide the pr into the documentation update and make a separate one for getting codecov to work.

Yes, please.

**Comment by kostis** [comment]
Can you please rebase on current `master` to resolve the conflicts?
(Commit https://github.com/proper-testing/proper/pull/163/commits/2e8a9427d44bca31aae67ab6d0bceb98004eeb68 should also be reverted.)
Thanks.

**Comment by TheGeorge** [comment]
I rebased and clean up the commit history for the PR

**Pull request merged**